### ISSUE TITLE
2.10.17 Bug: (On Import) ModuleNotFoundError: No module named 'resource' 

### ISSUE THREAD

**Issue by MrChadMWood** [PrefectHQ/prefect#10107]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Can not import Prefect in Jupyter Lab on Windows after upgrade to 2.10.17.

### Reproduction

```python3
`import prefect`
```

### Error

```python3
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[2], line 1
----> 1 import prefect

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\__init__.py:37
     24     warnings.warn(
     25         (
     26             "Prefect will drop support for Python 3.7 when it reaches end-of-life on 27"
   (...)
     32         stacklevel=2,
     33     )
     36 # Import user-facing API
---> 37 from prefect.states import State
     38 from prefect.logging import get_run_logger
     39 from prefect.flows import flow, Flow

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\states.py:14
     11 import pendulum
     12 from typing_extensions import TypeGuard
---> 14 from prefect.client.schemas import State as State
     15 from prefect.client.schemas import StateDetails, StateType
     16 from prefect.deprecated.data_documents import (
     17     DataDocument,
     18     result_from_state_with_data_document,
     19 )

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\client\schemas\__init__.py:4
      1 # Some objects are exported here for backwards compatibility.
      2 # In general, it is recommended to import schemas from their respective modules.
----> 4 from .actions import BlockTypeUpdate, StateCreate
      5 from .objects import (
      6     DEFAULT_BLOCK_SCHEMA_VERSION,
      7     BlockDocument,
   (...)
     19     Workspace,
     20 )
     21 from .responses import (
     22     OrchestrationResult,
     23     SetStateStatus,
   (...)
     26     StateRejectDetails,
     27 )

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\client\schemas\actions.py:9
      6 import jsonschema
      7 from pydantic import Field, root_validator, validator
----> 9 import prefect.client.schemas.objects as objects
     10 from prefect._internal.compatibility.experimental import experimental_field
     11 from prefect._internal.schemas.bases import ActionBaseModel

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\client\schemas\objects.py:20
     17 import pendulum
     18 from pydantic import Field, HttpUrl, conint, root_validator, validator
---> 20 from prefect._internal.schemas.bases import ObjectBaseModel, PrefectBaseModel
     21 from prefect._internal.schemas.fields import CreatedBy, DateTimeTZ, UpdatedBy
     22 from prefect._internal.schemas.validators import raise_on_name_with_banned_characters

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\_internal\schemas\bases.py:18
     15 from pydantic import BaseModel, Field, SecretField
     16 from pydantic.json import custom_pydantic_encoder
---> 18 from prefect._internal.compatibility.experimental import experiment_enabled
     19 from prefect._internal.schemas.fields import DateTimeTZ
     20 from prefect._internal.schemas.serializers import orjson_dumps

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\_internal\compatibility\experimental.py:19
     15 from typing import Any, Callable, Optional, Set, Type, TypeVar
     17 import pydantic
---> 19 from prefect.settings import PREFECT_EXPERIMENTAL_WARN, SETTING_VARIABLES, Setting
     20 from prefect.utilities.callables import get_call_parameters
     22 T = TypeVar("T", bound=Callable)

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\settings.py:70
     67 from pydantic import BaseSettings, Field, create_model, root_validator, validator
     68 from typing_extensions import Literal
---> 70 from prefect._internal.compatibility.deprecated import generate_deprecation_message
     71 from prefect.exceptions import MissingProfileError
     72 from prefect.utilities.names import OBFUSCATED_PREFIX, obfuscate

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\_internal\compatibility\deprecated.py:21
     18 import pydantic
     20 from prefect.utilities.callables import get_call_parameters
---> 21 from prefect.utilities.importtools import (
     22     AliasedModuleDefinition,
     23     AliasedModuleFinder,
     24     to_qualified_name,
     25 )
     27 T = TypeVar("T", bound=Callable)
     28 M = TypeVar("M", bound=pydantic.BaseModel)

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\utilities\importtools.py:17
     14 import fsspec
     16 from prefect.exceptions import ScriptError
---> 17 from prefect.utilities.filesystem import filename, is_local_path, tmpchdir
     20 def to_qualified_name(obj: Any) -> str:
     21     """
     22     Given an object, returns its fully-qualified name: a string that represents its
     23     Python import path.
   (...)
     29         str: the qualified name
     30     """

File ~\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\utilities\filesystem.py:5
      1 """
      2 Utilities for working with file systems
      3 """
      4 import os
----> 5 import resource
      6 import pathlib
      7 from contextlib import contextmanager

ModuleNotFoundError: No module named 'resource'
```
```


### Versions

```Text
Can not collect the desired information


C:\Users\prod_autos>prefect version

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Scripts\prefect.exe\__main__.py", line 4, in <module>
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\__init__.py", line 37, in <module>
    from prefect.states import State
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\states.py", line 14, in <module>
    from prefect.client.schemas import State as State
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\client\schemas\__init__.py", line 4, in <module>
    from .actions import BlockTypeUpdate, StateCreate
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\client\schemas\actions.py", line 9, in <module>
    import prefect.client.schemas.objects as objects
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\client\schemas\objects.py", line 20, in <module>
    from prefect._internal.schemas.bases import ObjectBaseModel, PrefectBaseModel
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\_internal\schemas\bases.py", line 18, in <module>
    from prefect._internal.compatibility.experimental import experiment_enabled
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\_internal\compatibility\experimental.py", line 19, in <module>
    from prefect.settings import PREFECT_EXPERIMENTAL_WARN, SETTING_VARIABLES, Setting
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\settings.py", line 70, in <module>
    from prefect._internal.compatibility.deprecated import generate_deprecation_message
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\_internal\compatibility\deprecated.py", line 21, in <module>
    from prefect.utilities.importtools import (
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\utilities\importtools.py", line 17, in <module>
    from prefect.utilities.filesystem import filename, is_local_path, tmpchdir
  File "C:\Users\prod_autos\AppData\Local\Programs\Python\Python311\Lib\site-packages\prefect\utilities\filesystem.py", line 5, in <module>
    import resource

ModuleNotFoundError: No module named 'resource'
```
```


### Additional context

https://pypi.org/project/prefect/2.10.17/

**Comment by jawnsy** [comment]
I think this should be resolved in our next release, which should be available soon.

The corresponding change was here: https://github.com/PrefectHQ/prefect/pull/10059

**Comment by serinamarie** [comment]
Hi @MrChadMWood, are you still seeing this issue in 2.10.18?

**Comment by jawnsy** [comment]
I believe this issue was resolved in #10059, so I'm closing this out. Please add a comment if you're still experiencing issues after upgrading to 2.10.18 - we'll be happy to assist with further investigation.

I've tested on Windows 11 in a VM that `prefect server start` works fine, as well as importing `prefect` via Jupyter (started from the same conda environment). I'm running Miniconda, which really helped with getting the environment set up, so I'd recommend that for Windows users.

![image](https://github.com/PrefectHQ/prefect/assets/52710/72740e55-f357-441e-800a-2c8420f3bfb3)
